packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -28,39 +28,44 @@ def check_tags(tags: dict, expected_tag_names: set):
 
 @pytest.mark.supportedwindows
 def test_metrics_agents_ping(dcos_api_session):
-    """ Test that the metrics service is up on masters.
+    """ Test that the metrics service is up on agents.
     """
-    for agent in dcos_api_session.slaves:
-        response = dcos_api_session.metrics.get('/ping', node=agent)
+    if dcos_api_session.slaves:
+        response = dcos_api_session.metrics.get('/ping', node=dcos_api_session.slaves[0])
         assert response.status_code == 200, 'Status code: {}, Content {}'.format(response.status_code, response.content)
         assert response.json()['ok'], 'Status code: {}, Content {}'.format(response.status_code, response.content)
         'agent.'
 
-    for agent in dcos_api_session.public_slaves:
-        response = dcos_api_session.metrics.get('/ping', node=agent)
+    if dcos_api_session.public_slaves:
+        response = dcos_api_session.metrics.get('/ping', node=dcos_api_session.public_slaves[0])
         assert response.status_code == 200, 'Status code: {}, Content {}'.format(response.status_code, response.content)
         assert response.json()['ok'], 'Status code: {}, Content {}'.format(response.status_code, response.content)
 
 
 @pytest.mark.supportedwindows
 def test_metrics_masters_ping(dcos_api_session):
-    for master in dcos_api_session.masters:
-        response = dcos_api_session.metrics.get('/ping', node=master)
+    if dcos_api_session.masters:
+        response = dcos_api_session.metrics.get('/ping', node=dcos_api_session.masters[0])
         assert response.status_code == 200, 'Status code: {}, Content {}'.format(response.status_code, response.content)
         assert response.json()['ok'], 'Status code: {}, Content {}'.format(response.status_code, response.content)
 
 
 def test_metrics_agents_prom(dcos_api_session):
     """Telegraf Prometheus endpoint is reachable on all agents."""
-    for agent in dcos_api_session.slaves:
-        response = dcos_api_session.session.request('GET', 'http://' + agent + ':61091/metrics')
+    if dcos_api_session.slaves:
+        response = dcos_api_session.session.request('GET', 'http://' + dcos_api_session.slaves[0] + ':61091/metrics')
+        assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
+
+    if dcos_api_session.public_slaves:
+        response = dcos_api_session.session.request(
+            'GET', 'http://' + dcos_api_session.public_slaves[0] + ':61091/metrics')
         assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
```

Comment:
More duplicated code here. This is another test where it probably makes sense to merge with its corresponding `test_metrics_masters_prom()`. They're all duplicating the same assertion logic.